webrtc/RTCSctpTransport-onstatechange.html

```diff
@@ -0,0 +1,56 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>RTCSctpTransport constructor</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="RTCPeerConnection-helper.js"></script>
+<script>
+'use strict';
+
+// Disable global timeout
+// IMPORTANT: You need to add a timeout to *every test* of this file!
+setup({ explicit_timeout: true });
+
+// Test is based on the following revision:
+// https://rawgit.com/w3c/webrtc-pc/1cc5bfc3ff18741033d804c4a71f7891242fb5b3/webrtc.html
+
+/*
+  4.4.2.  RTCPeerConnection.close
+    7.    Set the [[SctpTransportState]] slot of connection's [[SctpTransport]] to "closed".
+ */
+
+promise_test(async (t) => {
+  const resolver = new Resolver();
+  const pc1 = new RTCPeerConnection();
+  const pc2 = new RTCPeerConnection();
+  t.add_cleanup(() => pc1.close());
+  t.add_cleanup(() => pc2.close());
+
+  // We use the queue to hold back candidates, so no connection can be established
+  const pc1Queue = new IceCandidateQueue(pc1);
+  const pc2Queue = new IceCandidateQueue(pc2);
```

Comment:
This being the lone use of *IceCandidateQueue*, I propose using a promise as a simple queue here instead:
```js
promise_test(async (t) => {
  const pc1 = new RTCPeerConnection();
  const pc2 = new RTCPeerConnection();
  t.add_cleanup(() => pc1.close());
  t.add_cleanup(() => pc2.close());
  // We use a queue to hold back candidates, so no connection can be established
  let flush;
  const queue = new Promise(r => flush = r);
  pc1.onicecandidate = e => queue.then(() => pc2.addIceCandidate(e.candidate));
  pc2.onicecandidate = e => queue.then(() => pc1.addIceCandidate(e.candidate));

  // Negotiate SCTP transport
  pc1.createDataChannel('test');
  await doSignalingHandshake(pc1, pc2);

  for (const pc of [pc1, pc2]) {
    assert_equals(pc.sctp.state, 'connecting',
      'RTCSctpTransport should be in the connecting state');
  }
  flush(); // Let the peers connect
  await Promise.all([pc1, pc2].map(async pc => {
    await new Promise(r => pc.sctp.onstatechange = r);
    assert_equals(pc.sctp.state, 'connected');
  }));
}
```
This should also fix that the test doesn't bail before both `pc`s have had their `statechange` event tested.

Reply:
@lgrahl Odd, looks like github had some old review comments cached here. The last comment is the one I wanted you to look at.

Reply:
@youennf ^

Reply:
The first unresolved comment thread is the one about splitting this up. Presumably that's still happening? I just commented here because I couldn't find this test in the newer PRs. If we split it up and avoid modifying *exchangeIceCandidates* to take a queue, then I'm happy.